### DISCUSSION TITLE
Popover2 stopPropagation?

### DISCUSSION THREAD

**Discussion by synapse** [palantir/blueprint#5894] [Q&A] [answered]
Is there a way to stop the event propagation when using the Popover2.

I need something like this where when I click the button would not trigger the parent click action:

```
<div onClick={doSomething}>
    <Popover2 content={<Something />}>
        <button>open a menu</button>
    </Popover2>
</div>
```

Something like this breaks the popover from opening
```
<div onClick={doSomething}>
    <Popover2 content={<Something />}>
        <button onClick={event => event.stopPropagation()}>open a menu</button>
    </Popover2>
</div>
```

This works but it's an extra step to add an extra tag everywhere
```
<div onClick={doSomething}>
    <span onClick={event => event.stopPropagation()}>
        <Popover2 content={<Something />}>
            <button>open a menu</button>
        </Popover2>
    </span>
</div>
```

Is there a better way to do something like this?

**Comment by adidahiya** [comment, marked as answer]
Answered in https://github.com/palantir/blueprint/issues/5898#issuecomment-1410470283
